eva/experimental/ray/optimizer/rules/rules.py

```diff
@@ -104,7 +104,8 @@ def apply(self, before: LogicalGet, context: OptimizerContext):
         if before.target_list is None or not any(
             [isinstance(expr, FunctionExpression) for expr in before.target_list]
         ):
-            return scan
+            yield scan
```

Comment:
If there are no UDFs, we probably don't need ray. How about move this logic to the `check` function?

Reply:
I think we have to keep the logic here because it also needs to handle the sequential `LogicalGet` to `SeqScan` case. https://github.com/georgia-tech-db/eva/blob/1aac5cf09c99e992038cd8b4091332e233a48b2d/eva/optimizer/rules/rules_manager.py#L104

Reply:
oh I see. I thought SequentialLogicalGetToSeqScan and DistributedLogicalGetToSeqScan are two different rules.